web/src/hooks/useTempoInstance.ts

```diff
@@ -0,0 +1,47 @@
+import { useMemo } from 'react';
+import { useQueryParams } from './useQueryParams';
+
+/** a selected Tempo instance, including the tenant for multi-tenant instances */
+export interface TempoInstance {
+  namespace: string;
+  name: string;
+  tenant: string | undefined;
+}
+
+const tempoQueryParams = {
+  namespace: undefined,
+  name: undefined,
+  tenant: undefined,
+};
+
+export function useTempoInstance(): [
+  TempoInstance | undefined,
+  (tempo: TempoInstance | undefined) => void,
+] {
+  const [queryParams, setQueryParams] = useQueryParams(tempoQueryParams);
+
+  // either return a fully populated TempoInstance if all required params are set, or undefined
+  const tempo = useMemo(() => {
+    return queryParams.namespace && queryParams.name
+      ? {
+          namespace: queryParams.namespace,
+          name: queryParams.name,
+          tenant: queryParams.tenant,
+        }
+      : undefined;
+  }, [queryParams]);
+
+  const setTempo = (tempo: TempoInstance | undefined) => {
+    if (tempo) {
+      setQueryParams(tempo);
```

Comment:
would this create a new entry in the browser history? if so I think we should replace it instead as the user does not expect to go back and see just the instance changed, WDYT?

Reply:
Ok, I switched it.